packages/platforms/delta/x86-64/x86-64-delta-ak7448/onlp/builds/src/module/src/sysi.c

```diff
@@ -0,0 +1,348 @@
+int
+decide_percentage(int *percentage, int temper)
+{
+    int level;
+
+    if(temper <= 50)
+    {
+        *percentage = 50;
+        level = 1;
+    }
+    else if(temper > 50 && temper <= 55)
+    {
+        *percentage = 58;
+        level = 2;
+    }
+    else if(temper > 55 && temper <= 60)
+    {
+        *percentage = 65;
+        level = 3;
+    }
+    else if(temper > 60 && temper <= 65)
+    {
+        *percentage = 80;
+        level = 4;
+    }
+     else if(temper > 65)
+    {
+        *percentage = 100;
+        level = 5;
+    }
+    else
+    {
+        *percentage = 100;
+        level = 6;
+    }
+  
+    return level;
+}
+/******************************************************************************/
+
+
+int
+onlp_sysi_onie_data_get(uint8_t** data, int* size)
+{
+    uint8_t* rdata = aim_zmalloc(256);
+
+    if(onlp_file_read(rdata, 256, size, IDPROM_PATH) == ONLP_STATUS_OK) {
+        if(*size == 256) {
+            *data = rdata;
+            return ONLP_STATUS_OK;
+        }
+    }
+
+    aim_free(rdata);
+    *size = 0;
+    return ONLP_STATUS_E_UNSUPPORTED;
+}
+
+void
+onlp_sysi_onie_data_free(uint8_t* data)
+{
+    aim_free(data);
+}
+
+int
+onlp_sysi_platform_info_get(onlp_platform_info_t* pi)
+{
+    int sys_cpld_version = 0, cpld_a_version = 0, cpld_b_version = 0;
+
+    sys_cpld_version = dni_lock_cpld_read_attribute(SYS_CPLD_PATH,SYS_VERSION_REG);
+    cpld_a_version = dni_lock_cpld_read_attribute(CPLD_A_PATH,CPLD_A_VERSION_REG);
+    cpld_b_version = (dni_lock_cpld_read_attribute(CPLD_B_PATH,CPLD_B_VERSION_REG) & 0x7);
+
+    pi->cpld_versions = aim_fstrdup("SYSTEM-CPLD = %d, CPLD-A = %d, CPLD-B = %d", sys_cpld_version, cpld_a_version, cpld_b_version);
+
+    return ONLP_STATUS_OK;
+}
+
+void
+onlp_sysi_platform_info_free(onlp_platform_info_t* pi)
+{
+    aim_free(pi->cpld_versions);
+}
+
+int
+onlp_sysi_oids_get(onlp_oid_t* table, int max)
+{
+    int i;
+    onlp_oid_t* e = table;
+    memset(table, 0, max*sizeof(onlp_oid_t));
+
+    /* 7 Thermal sensors on the chassis */
+    for (i = 1; i <= NUM_OF_THERMAL_ON_BOARDS; i++)
+    {
+        *e++ = ONLP_THERMAL_ID_CREATE(i);
+    }
+
+    /* 6 LEDs on the chassis */
+    for (i = 1; i <= NUM_OF_LED_ON_BOARDS; i++)
+    {
+        *e++ = ONLP_LED_ID_CREATE(i);
+    }
+
+    /* 7 Fans on the chassis */
+    for (i = 1; i <= NUM_OF_FAN_ON_FAN_BOARD; i++)
+    {
+        *e++ = ONLP_FAN_ID_CREATE(i);
+    }
+
+    /* 1 PSUs on the chassis */
+    for (i = 1; i <= NUM_OF_PSU_ON_PSU_BOARD; i++)
+    {
+        *e++ = ONLP_PSU_ID_CREATE(i);
+    }
+
+    return 0;
+}
+
+int
+onlp_sysi_platform_manage_fans(void)
+{
+    int i = 0;
+    int new_percentage;
+    int highest_temp = 0;
+    onlp_thermal_info_t thermal[NUM_OF_THERMAL_ON_BOARDS];
+
+    /* Get current temperature */
+    if (onlp_thermali_info_get(ONLP_THERMAL_ID_CREATE(THERMAL_CPU_CORE),        &thermal[0]) != ONLP_STATUS_OK ||
```

Comment:
The onlp_thermali_info_get() in onlp_sysi_platform_manage_*  read  the attribute by  dni_i2c_lock_read_attribute() function.
https://github.com/delta-JohnsonLu/OpenNetworkLinux/blob/b92a1c57e114f2a6e321bddaf370ce5fc60d7eed/packages/platforms/delta/x86-64/x86-64-delta-ak7448/onlp/builds/src/module/src/platform_lib.c#L130